lib/facade_converter.ts

```diff
@@ -170,9 +170,16 @@ export class FacadeConverter extends base.TranspilerBase {
     'Array.concat': (c: ts.CallExpression, context: ts.Expression) => {
       this.emit('new List . from (');
       this.visit(context);
-      this.emit(') .. addAll (');
-      this.visit(c.arguments[0]);
       this.emit(')');
+      c.arguments.forEach(arg => {
+        var symbol = this.tc.getTypeAtLocation(arg).getSymbol();
+        if (!symbol || this.tc.getFullyQualifiedName(symbol) !== 'Array') {
```

Comment:
This is not quite enough. You'll also have to make sure that this is the `Array` symbol that's the globally defined `Array`, the one from `lib.d.ts`. So check if the location where `symbol` is declared is in that particular file.

I don't quite know how to explain this better, I feel like I'm not adding much to what I wrote before. If it's ok for you I can take over the PR?